lib/sycamore/sycamore/utils/fileformat_tools.py

```diff
@@ -0,0 +1,44 @@
+import subprocess
+from tempfile import NamedTemporaryFile
+
+from sycamore.data import Document
+
+
+def convert_file_to_pdf(doc: Document) -> Document:
+    """
+    Utility to convert binary_representations into different file formats. Uses LibreOffice as the conversion engine.
+
+    Note: LibreOffice currently requires manual installation based on your platform.
+    """
+
+    def run_libreoffice(source_path, output_path):
+        subprocess.run(
+            [
+                "libreoffice",
+                "--headless",
+                "--convert-to",
+                "pdf",
+                source_path,
+                "--outdir",
+                output_path,
+            ]
+        )
+
+    assert doc.binary_representation is not None
+
+    current_filetype = doc.properties.get("filetype")
+    assert current_filetype is not None, "Document requires properties.filetype"
+
+    with NamedTemporaryFile(suffix=f".{current_filetype.split('/')[-1]}") as temp_file:
+        temp_file.write(doc.binary_representation)
+        temp_file.flush()
+
```

Comment:
```
temp_path = Path(temp_file)
run_libreoffice(temp_file.name, temp_path.parent)
with open(temp_path.parent / temp_path.stem + ".pdf", "rb") as processed_file:
    doc.binary_representation = processed_file.read()
```

Reply:
the file extension when reading .docx was causing the Path lib to break (it includes multiple docs, e.g. 'application/vnd.openxmlformats-officedocument.wordprocessingml.document'), so I needed to do it this way. 
I can choose to drop the file extensionbut figured its more useful when debugging